policybot/dashboard/topics/postsubmit/analysis.html

```diff
@@ -17,84 +17,94 @@
-              <tbody class="collapsed"> 
-                <td class="subtd">
-                {{ range .SubLabel.LabelEnv}}
-                  <table class="subtable">
-                    <tr>
-                      <thead>
-                        <td>&nbsp &nbsp {{ .Label }}</td>
-                        {{ range .EnvCount }}
-                        <td>{{ . }}</td>
-                        {{ end }}
-                      </thead>
-                    </tr>
-                    <tbody class="collapsed">
-                      <td class="subtd">
-                        {{ range .SubLabel.LabelEnv}}
-                          <table class="subtable">
-                            <tr>
-                              <thead>
-                                <td>&nbsp &nbsp &nbsp {{ .Label }}</td>
-                                {{ range .EnvCount }}
-                                <td>{{ . }}</td>
-                                {{ end }}
-                              </thead>
-                            </tr>
-                            <tbody class="collapsed">
-                              {{ range .SubLabel.LabelEnv}}
-                                <tr>
-                                  <td>&nbsp &nbsp &nbsp &nbsp {{ .Label }}</td>
-                                  {{ range .EnvCount }}
-                                  <td>{{ . }}</td>
-                                  {{ end }}
-                                </tr>
-                              {{ end }}
-                            </tbody>
-                          </table>
-                        {{ end }}
-                      </td>       
-                    </tbody>
-                  </table>
-                {{ end }}
-                </td>                
-              </tbody>
-            </table>  
-          {{ end }}
-          </td>
-        </tbody>
+        {{template "innerlayer" (wrap .SubLabel $depth)}}      
       </table>
     {{ end }}
   </tbody>
 </table>
+<br>
+<p> View TestNames</p>
+<table>
+  <thead>
+    <tr>
+      <th>TestName</th>
+    </tr>
+  </thead>
+  <tbody>
+    {{range $testname := .TestNameByEnvLabels}}
+      <tr>
+        <td>{{ $testname.TestOutcomeName }} 
+          <a href="https://prow.istio.io/view/gcs/istio-prow/logs/{{ $testname.TestName }}/{{ $testname.RunNumber }}">Prow Link</a></td>
+      </tr>
+    {{ end }}
+  </tbody>
+</table>
+
+{{ define "innerlayer" }}
+<tbody class="collapsed">
+  <td class="subtd">
+    {{$depth := .Depth}}
+    {{ range .LabelEnv}}
+      <table class="subtable">
+        <thead>
+          <tr>
+            <td class="pivot">
+              {{range slice $depth}}&nbsp;{{end}}{{ .Label }}
+            </td>
+            {{ range .EnvCount }}
+            <td>{{ . }} <button class="{{$depth}}" type="button">View All Test</button></td>
+            {{ end }}
+          </tr>
+        </thead>
+        {{ if .SubLabel.LabelEnv }}
+          {{template "innerlayer" (wrap .SubLabel $depth)}}
```

Comment:
sorry to ask for another change, but is there a way we can make this a bit more generic? Something like creating a dictionary dynamically, e.g. (dict "sublabel" .SubLabel "depth" .Depth)

Reply:
I changed to the map interface, but since function needs to declare all the inputs and their type, it can't be totally dynamic.